code/modules/library/lib_machines.dm

```diff
@@ -374,6 +377,18 @@
 			var/obj/machinery/libraryscanner/scan = get_scanner()
 			data["has_scanner"] = !!(scan)
 			data["has_cache"] = !!(scan?.cache)
+
+			var/cooldown_number = (COOLDOWN_TIMELEFT(src, newscaster_cooldown))
+			var/active_newscaster_cooldown = COOLDOWN_FINISHED(src, newscaster_cooldown)
+			data["active_newscaster_cooldown"] = active_newscaster_cooldown
+			var/tooltip_message = ""
+			if(active_newscaster_cooldown)
+				tooltip_message = "Upload your book to the newscaster's book club channel."
+			else
+				tooltip_message = "Please wait [DisplayTimeText(cooldown_number)] before uploading your book."
+
+			data["upload_tooltip_message"] = tooltip_message
```

Comment:
```suggestion
			if(active_newscaster_cooldown)
				data["upload_tooltip_message"] = "Upload your book to the newscaster's book club channel."
			else
				data["upload_tooltip_message"] = "Please wait [DisplayTimeText(cooldown_number)] before uploading your book."
```

Reply:
Actually, this text should be baked into the UI, and use `active_newscaster_cooldown` variable to show one or the other.